src/google/protobuf/compiler/objectivec/objectivec_message.cc

```diff
@@ -504,7 +518,7 @@ void MessageGenerator::GenerateSource(io::Printer* printer) {
     printer->Print(
         vars,
         "    GPBDescriptor *localDescriptor =\n"
-        "        [GPBDescriptor allocDescriptorForClass:[$classname$ class]\n"
+        "        [GPBDescriptor allocDescriptorForClass:self\n"
```

Comment:
Please hold off on this change, if someone subclassed a message, this will latch onto that class if it is the first instance used for a message. 

To do this, we need to first tag everything as not subclass-able, to sure this change doesn't go wrong (and making that a distinct commit with this will be cleaner).

Reply:
Fixed